spec/features/user_can_renew_book_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+require 'date'
+
+feature 'User can renew a checked out book' do
+  scenario 'sucessfully' do
+    patron = create(:patron)
+    book = create(:book)
+    checkout = create(:checkout, user: patron, book: book)
+
+    visit profile_path(as: patron)
+
+    click_on 'Renew'
+
+    checkout.reload
+
+    original_checkout_date = book.checkouts.first.created_at
+    expected_due_date_after_renewal = format_date(original_checkout_date + (2 * Checkout::CHECKOUT_PERIOD_IN_DAYS.days))
+    due_date_after_renewal = format_date(book.checkouts.last.due_on)
+    expect(page).to have_due_date(due_date_after_renewal)
+    expect(due_date_after_renewal).to eq(expected_due_date_after_renewal)
+    expect(page).to have_content(t('checkouts.update.successful_message'))
+  end
+
+  scenario 'prior fines carry over when an overdue book is renewed' do
+    patron = create(:patron)
+    book = create(:book)
+    checkout = create(:checkout, user: patron, book: book)
+
+    date_one_day_overdue = date_when_book_is_one_day_overdue(DateTime.now)
+    Timecop.travel(date_one_day_overdue) do
+      visit profile_path(as: patron)
+
+      click_on 'Renew'
+    end
+
+    checkout.reload
+
+    # one_day_overdue = 1
```

Comment:
@paulcsmith, This commented out method of calculating the fine matches the old approach where the fine was "bubbled up" from each checkout and only ever lived on one checkout. I am thinking of a way to duplicate this test so it fits the current approach of having each checkout reflect the fine associated with it's checkout period.